.github/workflows/test.yaml

```
@@ -1132,23 +1130,73 @@ jobs:
       - name: Install Dependencies
         run: pnpm install
 
-      - name: test ${{ matrix.platform }}
-        id: run-test
+      - name: test DP1 ${{ matrix.platform }}
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 60
+          max_attempts: 3
+          retry_wait_seconds: 30
+          command: bash .github/scripts/test-project.sh ${{ github.job }} ${{ matrix.platform }}
+        env:
+          DATAPROXY_FLAVOR: 'DP1'
+          PRISMA_GENERATE_FLAG: '--data-proxy'
+          DATAPROXY_COMMON_URL: ${{ secrets.DATAPROXY_COMMON_URL }}
+          ITX_PDP_MONGODB_DATABASE_URL: ${{ secrets.ITX_PDP_MONGODB_DATABASE_URL }}
+          ITX_PDP_MONGODB: ${{ secrets.ITX_PDP_MONGODB }}
+          ITX_PDP_MYSQL_DATABASE_URL: ${{ secrets.ITX_PDP_MYSQL_DATABASE_URL }}
+          ITX_PDP_MYSQL: ${{ secrets.ITX_PDP_MYSQL }}
+          ITX_PDP_POSTGRESQL_DATABASE_URL: ${{ secrets.ITX_PDP_POSTGRESQL_DATABASE_URL }}
+          ITX_PDP_POSTGRESQL: ${{ secrets.ITX_PDP_POSTGRESQL }}
+          ITX_PDP_COCKROACHDB_DATABASE_URL: ${{ secrets.ITX_PDP_COCKROACHDB_DATABASE_URL }}
+          ITX_PDP_COCKROACHDB: ${{ secrets.ITX_PDP_COCKROACHDB }}
+
+      - name: test DP2 ${{ matrix.platform }}
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 60
```

```suggestion
          timeout_minutes: 15
```
Would be better, I think?